src/traces/sankey/attributes.js

```diff
@@ -89,6 +89,16 @@ var attrs = module.exports = overrideAll({
             role: 'info',
             description: 'The shown name of the node.'
         },
+        groups: {
+            valType: 'data_array',
```

Comment:
It might be better to declare this as an `info_array` with `dimensions: 2` similar to 

https://github.com/plotly/plotly.js/blob/bb1239a31a56d72ec664091ca7e1c0b42adb8b2f/src/components/grid/index.js#L54-L69

that way wouldn't need to worry about its structure during `sankey/calc.js`.

Reply:
I didn't realize we had `info_array`. Did you mean `dimensions: 1`? 

For example: the following seem to work fine:
```javascript
        groups: {
            valType: 'info_array',
            dimensions: 1,
            freeLength: true,
            items: {valType: 'data_array', dflt: [], editType: 'calc'},
            role: 'calc',
            description: [
                'Groups of nodes.',
                'Each group is defined by an array with the indices of the nodes it contains.',
                'Multiple groups can be specified.'
            ].join(' ')
        },
```

Reply:
Why do the inner items need to be `data_array`?

Reply:
Oh I see! The following works fine too:
```javascript
        groups: {
            valType: 'info_array',
            dimensions: 2,
            freeLength: true,
            items: {valType: 'number', editType: 'calc'},
            role: 'calc',
            description: [
                'Groups of nodes.',
                'Each group is defined by an array with the indices of the nodes it contains.',
                'Multiple groups can be specified.'
            ].join(' ')
        },
```

Reply:
> The following works fine too:

Nice. I'd vote for this configuration. I don't think `groups` is really a "data" array.

Reply:
Done in 1cd206f7fdba75b010650331fc1f7a726ec488df